### PR TITLE
a scenario on upgrading and migrating contracts

### PR DESCRIPTION
**Pull request by ghost**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: upgrading/upgrade-runnning-applications/2_new_feature.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+data with a `nickname` field, such that the beginning of the template looks like
+
+<pre class="file" data-target="clipboard">
+template User
+  with
+    username: Party
+    following: [Party]
+    nickname: Optional Text
+  where
+    ...
```
Comment (nemanja-da) [inline upgrading/upgrade-runnning-applications/2_new_feature.md]:
```suggestion
```

If you copy it like this id adds the three dots and the IDE complains that it's an error.

File: upgrading/upgrade-runnning-applications/1_deploy.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+To create a few users, posts and comments of the social network run the `initialize` script in the
+`forum-0.1.0.dar` package:
+
+```
+daml script --dar forum-0.1.0.dar --script-name Init:initialize --ledger-host localhost --ledger-port 6865 --wall-clock-time
+```{{execute T2}}
+
+Nice, your social network is running and users are happily posting and commenting in the forum
+(look at the Init.daml module in the forum source code to see what). In the next step we'll upgrade
```
Comment (nemanja-da) [inline upgrading/upgrade-runnning-applications/1_deploy.md]:
```suggestion
(look at the `../forum/Init.daml`{{open}} module in the forum source code to see what). In the next step we'll upgrade
```

File: upgrading/upgrade-runnning-applications/4_migrate_contract.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+rollbackPost V2.Forum.Post{..} = V1.Forum.Post with user = rollbackUser user, ..
+
+upgradeComment : V1.Forum.Comment -> ContractId V2.Forum.Post -> V2.Forum.Comment
+upgradeComment V1.Forum.Comment{..} newPostCid
+  = V2.Forum.Comment with commenter = upgradeUser commenter, post = newPostCid, ..
+
+rollbackComment : V2.Forum.Comment -> ContractId V1.Forum.Post -> V1.Forum.Comment
+rollbackComment V2.Forum.Comment{..} oldPostCid
+  = V1.Forum.Comment with commenter = rollbackUser commenter, post = oldPostCid, ..
+
```
Comment (nemanja-da) [inline upgrading/upgrade-runnning-applications/4_migrate_contract.md:37]: Can you explain here why upgrade and rollback are needed and what each does? I know it's trivial, but we have to spell it out to the user.

File: upgrading/upgrade-runnning-applications/4_migrate_contract.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        archive commentCid
+        create $ rollbackComment comment oldPostCid
+</pre>
+
+1. The `MigrateUser` template offers every user of the mini social network of the
+   create-daml-app-0.1.0 package to migrate his `User`, `Post` and `Comment` contracts to the
+   updated version of the packages.
+1. We use qualified imports to avoid name clashes of the two imported modules. Note how we can
+   import the two equally named modules of the two packages with the given prefix we added in the
+   `daml.yaml` file in the previous step.
```
Comment (nemanja-da) [inline upgrading/upgrade-runnning-applications/4_migrate_contract.md]:
```suggestion
   `../migration/daml.yaml`{{open}} file in the previous step.
```
Just wanna give the user a shortcut so s/he can open the file and take a look again.

File: upgrading/upgrade-runnning-applications/1_deploy.md (hunk 2 above)
Comment (ghost) [reply upgrading/upgrade-runnning-applications/1_deploy.md]: changed.

File: upgrading/upgrade-runnning-applications/4_migrate_contract.md (hunk 4 above)
Comment (ghost) [reply upgrading/upgrade-runnning-applications/4_migrate_contract.md]: changed.

File: upgrading/upgrade-runnning-applications/4_migrate_contract.md (hunk 3 above)
Comment (ghost) [reply upgrading/upgrade-runnning-applications/4_migrate_contract.md:37]: It's explained in the last bullet point on that page, would you like to move it further up?

File: upgrading/upgrade-runnning-applications/2_new_feature.md (hunk 1 above)
Comment (ghost) [reply upgrading/upgrade-runnning-applications/2_new_feature.md]: removed it.

**Review by nemanja-da — APPROVED** [review]
TYVM @robin-da

File: upgrading/upgrade-runnning-applications/2_new_feature.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+and open `daml/User.daml`{{open}} in the IDE tab and wait for it to load. We'll extend the `User`
+data with a `nickname` field, such that the beginning of the template looks like
+
+<pre class="file" data-target="clipboard">
+template User where
+    username: Party
+    following: [Party]
+    nickname: Optional Text
+  where
+</pre>
```
Comment (nemanja-da) [inline upgrading/upgrade-runnning-applications/2_new_feature.md:17]:
```suggestion
template User with
    username: Party
    following: [Party]
    nickname: Optional Text
  where
</pre>
```

**Pull request merged**